### PR TITLE
Software Recruitment Challenge

### PR DESCRIPTION
**Pull request by Burke-Daniel**
## Overview
Initial pull request and implementation of software recruitment challenge. The idea is to have the recruits set up their development environment (text editor and git), clone the repo, make a branch, complete the coding part of the challenge, and submit a PR to this repo.

## Code
The code part of the code challenge consists of 3 classes, 1 of which the recruit is to complete.
 - The GPS class allows the user to obtain the boring machine's current position
 - The Steering class allows the user to interact with the "actuator" that controls the borer's pitch and yaw, as well as its forward movement.
 - The Navigation class will be implemented by the recruits.

@colton-smith feel free to change/add to this, I wrote all of this very fast and there may be mistakes or parts that could be a little more clear

### CODE REVIEW

File: software-challenge/navigation.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        pass
+
+    """ Navigate to the desired position from the current position
+    
+    Based on the current position tuple, compared to the desired position tuple,
+    make decisions on steering, and ensure that the actuations are successful
+
+    Returns: True if actuation requests were successful, False if not
+    Note: It may be good to notify the user if something unexpected happens!
+    """
```
Comment (colton-smith) [inline software-challenge/navigation.py:67]: These various docstrings should go underneath the function prototype :) I'll fix this up

File: software-challenge/navigation.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
+"""
+You will be implementing a hypothetical navigation class,
+which keeps track of the boring machine's current position, its desired position,
+and makes any adjustments necessary to keep the machine on track.You will make the
+assumption that two other interfaces are written to make your life easier, the GPS
+class will interface with our onboard GPS sensor to get thecurrent position, and
```
Comment (colton-smith) [inline software-challenge/navigation.py:6]: thecurrent -> the current

File: software-challenge/navigation.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        # These will be tuples of the form (x, y, z)
+        self.current_position = None
+        self.desired_position = None
+    
+    """ Sets the desired position that the borer will attempt to move to
+    
+    Note: assume the user will pass in the desired_position parameter when using
+          the interface
+    """
+    def set_desired_position(self, desired_position):
```
Comment (colton-smith) [inline software-challenge/navigation.py:53]: Should add a comment that indicates the user should implement the method.

**Review by colton-smith — CHANGES_REQUESTED** [review]
Looks mostly good, just have to move the docstrings to the correct location (you've been doing too much c++ hahahah) and possibly add some notes about classes - just remembered ENGI1020 doesnt cover OOP so we may have to just add a little blurb in the readme about classes (just attributes and methods basically). The other option is too just ask to them implement some functions that take a GPS object as a param, although its good for them to see classes as well! 

Since you're off in old trin town I'll fix this up !

**Comment by colton-smith** [comment]
I updated all of the docstrings, and added to the readme, ready for re-review!

File: fall-2020/README.md [hunk 4]
```diff
@@ -24,3 +24,62 @@ Finally, for the actual coding part of the challenge, you will have to implement
...
+
+
+## Appendix
+As many of the applicants will not have much experience with OOP (what is that ???), so here are some quick notes! However, I would encourage you to google and or read ahead in your engi 3891 course notes 
+to learn more, these notes are very sparse.
+
+Main Points:
+* OOP - Object Oriented Programming
+* Keep data and ways to interact with that data together 
+* A class is like a blueprint for an object (oftem classes represent real world objects)
```
Comment (Burke-Daniel) [inline fall-2020/README.md:38]: lil typo here

File: fall-2020/README.md [hunk 5]
```diff
@@ -24,3 +24,62 @@ Finally, for the actual coding part of the challenge, you will have to implement
...
+
+    def park()
+
+    def autopilot():
+
+    def activate_cruise():
+
+    def change_tire():
+
+    def activate_vtec():
```
Comment (Burke-Daniel) [inline fall-2020/README.md:70]: lol when the vtec kicks in 😲

File: fall-2020/README.md [hunk 6]
```diff
@@ -24,3 +24,62 @@ Finally, for the actual coding part of the challenge, you will have to implement
...
+
+To create an instance of the class, you simply call the constructor and provide the following arguements" make, model, max_speed, tire_size, is_coupe
```
Comment (Burke-Daniel) [inline fall-2020/README.md:72]: lil typo here too

File: fall-2020/README.md [hunk 7]
```diff
@@ -24,3 +24,62 @@ Finally, for the actual coding part of the challenge, you will have to implement
...
+You can then view the value of the variables in the class, and call methods:
+
+car_name = Civic.name
+car_max_speed = Civic.maxspeed
+
+Civic.drive_forward()
+Civic.park()
+Civic.activate_vtec()
+```
+# GLHF !
```
Comment (Burke-Daniel) [inline fall-2020/README.md:85]: We should see if the mechs can give the boring machine vtec

**Review by Burke-Daniel — COMMENTED** [review]
Aside from fixing my C++ docstrings, there are some nice documentation refinements here, good stuff.

**Comment by Burke-Daniel** [comment]
Just need an approval from you on this then will merge

**Review by colton-smith — APPROVED** [review]
GTM!

**Pull request merged**